### ISSUE TITLE
Output mapping functionality

### ISSUE THREAD

**Issue by jbrantly** [adriantoine/enzyme-to-json#48]
Would you be open to the addition of some form of node mapping functionality added on top of the existing serialization? Here are some use cases that I've run into where something like this could be really useful:

1. Unwrapping HOC components so that the correct component name can be displayed. It's pretty easy in Enzyme to unwrap the root component but much more difficult (and likely unnecessary) to unwrap descendants. If these could be unwrapped for snapshot output though it would make the output much prettier.
2. Ignoring specific props. In some cases you may not want to pretty-print complex object graphs being passed through props.

I'm sure there are all sorts of other use-cases that people could come up with. It would be awesome if you could specify a function somewhere that would be called for each transformed node and would have access to the raw node as well as the default JSON output which it could transform.

If this is something you would be amenable to I'd be happy to open a PR.

**Comment by adriantoine** [comment]
Hi @jbrantly!

I am open to any kind of improvement, but I don't think I understand your use cases clearly. Can you provide code examples? As far as I know, if you can do something with Enzyme, you can do it with `enzyme-to-json` 🤔

Thanks!

**Comment by sboudrias-aa** [comment]
@adriantoine Hey, I think this would be really useful for people using CSS-in-JS tools. A lot of them are creating a lot of unnecessary noise in the snapshot files.

For example, with styled-jsx:

```
exports[`<Overlay.Control> click trigger 1`] = `
<div
  className="OverlayControl"
  data-jsx={3299129060}
  onClick={[Function]}
>
  children
  <_class
    css=".OverlayControl[data-jsx=\\"3299129060\\"] {display: inline-block;position: relative;}"
    styleId={4150426368}
  />
  <_class
    css="
                  .OverlayControl .Overlay {
                      display: none;
                  }
              "
    styleId={3597040161}
  />
</div>
`;
```

Or with glamorous:

```
exports[`<Button> renders text 1`] = `
<button
  className=""
  data-css-1qmv7zm=""
>
  My Button
</button>
`;
```

Having less verbose snapshot would certainly be welcomed by a lot of these tools users. Right now they make snapshot a pain to review.

**Comment by adriantoine** [comment]
@sboudrias-aa it does look like an interesting feature. @jbrantly has started a PR a while ago, I'll see if I can implement it myself.

Thanks for your feedback.

**Comment by pedrottimark** [comment]
@adriantoine Another way to ignore specific props is by [toMatchObject](http://facebook.github.io/jest/docs/expect.html#tomatchobjectobject) assertion in Jest, illustrated by the description of a sample project. The best place to start reading is here:

https://github.com/pedrottimark/whimper#interlude-about-snapshot-testing

If you are willing for the API of `enzyme-to-json` to include `mountToDeepJson` and `mountToShallowJson` then I can make a pull request for your review.

**Comment by adriantoine** [comment]
Fixed by #56

**Comment by adriantoine** [comment]
I'm reopening that as it's not really what was implemented in #56, I'll work on a solution for `v3.0` (see #67)

**Comment by adriantoine** [comment]
Ok well I'm closing this again 😁 Just to let you know that the mapping feature will be in `3.0` all the details there: https://github.com/adriantoine/enzyme-to-json/issues/67#issuecomment-330368493
